plugins/providers/gcs/client.go

```diff
@@ -84,3 +80,48 @@ func (c *gcsClient) RevokeBucketAccess(ctx context.Context, b Bucket, identity s
 
 	return nil
 }
+
+func (c *gcsClient) ListAccess(ctx context.Context, resources []*domain.Resource) (domain.MapResourceAccess, error) {
+	result := make(domain.MapResourceAccess)
+
+	for _, resource := range resources {
+		var accessEntries []domain.AccessEntry
+
+		bucket := c.client.Bucket(resource.URN)
+		policy, err := bucket.IAM().Policy(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("Bucket(%q).IAM().Policy: %w", resource.URN, err)
+		}
+
+		for _, role := range policy.Roles() {
+			for _, member := range policy.Members(role) {
+				var accountType, accountID string
+				m := strings.Split(member, ":")
+				if len(m) == 0 || len(m) > 2 {
+					return nil, fmt.Errorf("invalid bucket access member signature %q", member)
+				} else if len(m) == 2 {
+					accountID = m[1]
+				}
+				accountType = m[0]
```

Comment:
let's move this to separate function to make it easier to read and less responsibility on this function